services/marmotgraph-core-api/src/main/java/org/marmotgraph/core/controller/CoreCacheController.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 - 2021 Swiss Federal Institute of Technology Lausanne (EPFL)
+ * Copyright 2021 - 2022 EBRAINS AISBL
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This open source software code was developed in part or in whole in the
+ * Human Brain Project, funded from the European Union's Horizon 2020
+ * Framework Programme for Research and Innovation under
+ * Specific Grant Agreements No. 720270, No. 785907, and No. 945539
+ * (Human Brain Project SGA1, SGA2 and SGA3).
+ */
+
+package org.marmotgraph.core.controller;
+
+import org.marmotgraph.commons.AuthContext;
+import org.marmotgraph.commons.exception.UnauthorizedException;
+import org.marmotgraph.commons.permission.Functionality;
+import org.marmotgraph.commons.permissions.controller.Permissions;
+import org.springframework.cache.CacheManager;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@Component
+public class CoreCacheController {
+
+    private static final String NO_RIGHTS_TO_FLUSH_CACHE = "You are not allowed to flush cache";
+
+    private final AuthContext authContext;
+    private final Permissions permissions;
+    private final CacheManager cacheManager;
+
+    public CoreCacheController(AuthContext authContext, Permissions permissions, CacheManager cacheManager) {
+        this.authContext = authContext;
+        this.permissions = permissions;
+        this.cacheManager = cacheManager;
+    }
+
+    public List<String> clearKeys(List<String> keys) {
+
+        if (!permissions.hasGlobalPermission(authContext.getUserWithRolesWithoutTermsCheck(), Functionality.CACHE_FLUSH)) {
+            throw new UnauthorizedException(CoreCacheController.NO_RIGHTS_TO_FLUSH_CACHE);
+        }
+
+        List<String> keysFlushed = new ArrayList<>();
+
+        cacheManager.getCacheNames().forEach(cacheName -> {
+            if( keys.contains(cacheName) ) {
+                Objects.requireNonNull(cacheManager.getCache(cacheName)).clear();
+                keysFlushed.add(cacheName);
+            }
+        });
+
+        return keysFlushed;
+    }
+
+    public List<String> clearAllKeys() {
+
+        if (!permissions.hasGlobalPermission(authContext.getUserWithRolesWithoutTermsCheck(), Functionality.CACHE_FLUSH)) {
```

Comment:
I think this additional check is not needed since you call "clearKeys" afterwards which already does exactly the same check

Reply:
Thanks for the comment! I agree with what you wrote but I think the entire "clearAllKeys" method should be protected for any extra admin operation that happens before "clearKeys" (e.g. cacheManager calls) :+1: